exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/TransactionScope.kt

```diff
@@ -5,31 +5,18 @@ import org.jetbrains.exposed.sql.Transaction
 import kotlin.properties.ReadWriteProperty
 import kotlin.reflect.KProperty
 
-@Suppress("UNCHECKED_CAST")
-fun <T:Any> transactionScope(init: Transaction.() -> T) = TransactionStore(init) as ReadWriteProperty<Any?, T>
-fun <T:Any> nullableTransactionScope() = TransactionStore<T>()
-
-class TransactionStore<T:Any>(val init: (Transaction.() -> T)? = null) : ReadWriteProperty<Any?, T?> {
+fun <T:Any> transactionScope(init: Transaction.() -> T): ReadWriteProperty<Any?, T> = TransactionScope(init)
 
+private class TransactionScope<T: Any>(
+    private val init: Transaction.() -> T
+) : ReadWriteProperty<Any?, T> {
     private val key = Key<T>()
 
-    @Suppress("UNCHECKED_CAST")
-    override fun getValue(thisRef: Any?, property: KProperty<*>): T? {
-        val currentOrNullTransaction = TransactionManager.currentOrNull()
-        return currentOrNullTransaction?.getUserData(key)
-            ?: init?.let {
-                val value = currentOrNullTransaction!!.it()
```

Comment:
`currentOrNullTransaction!!` means that a transaction is required, so I've changed the code to use `TransactionManager.current()`.  Removing `nullableTransactionScope` means that `init` can now always return a non-nullable value, which is in line with how `transactionScope` is used.